code/game/machinery/roulette_machine.dm

```diff
@@ -277,7 +295,7 @@
 	if(locked)
 		return
 	locked = TRUE
-	var/stolen_cash = my_card.registered_account.account_balance * percentage
+	var/stolen_cash = my_card?.registered_account?.account_balance * percentage
```

Comment:
doesn't this become null * number? I don't think you should be doing that.

Reply:
Byond evaluates nulls as 0 in numeric contexts (such as math operations).